gui/app.py

```diff
@@ -145,6 +157,50 @@ def connect_signals(self):
+        self.main_tab.widgets.video_area.big_video_changed_signal.connect(self.main_tab.show_prompts_for_cam)
+
+        for relay_button in (
+            self.main_tab.widgets.front_deployer_button,
+            self.main_tab.widgets.front_claw_button,
+            self.main_tab.widgets.back_deployer_button,
+            self.main_tab.widgets.back_claw_button,
+            self.main_tab.widgets.magnet_button,
+            self.main_tab.widgets.lights_button,
+        ):
+            self.vehicle.armed_signal.connect(relay_button.enable_click)
+            self.vehicle.disarmed_signal.connect(relay_button.on_disarm)
+            self.vehicle.disconnected_signal.connect(relay_button.on_disarm)
+
+        # Connect the manipulator buttons to their manipulators
+        self.main_tab.widgets.front_deployer_button.state_change_signal.connect(
+            lambda state: self.vehicle.set_relay(Relay.PVC_FRONT, state))
+        self.main_tab.widgets.front_claw_button.state_change_signal.connect(
+            lambda state: self.vehicle.set_relay(Relay.CLAW_FRONT, state))
+        self.main_tab.widgets.back_deployer_button.state_change_signal.connect(
+            lambda state: self.vehicle.set_relay(Relay.PVC_BACK, state))
+        self.main_tab.widgets.back_claw_button.state_change_signal.connect(
+            lambda state: self.vehicle.set_relay(Relay.CLAW_BACK, state))
+        self.main_tab.widgets.magnet_button.state_change_signal.connect(
+            lambda state: self.vehicle.set_relay(Relay.MAGNET, state))
+        self.main_tab.widgets.lights_button.state_change_signal.connect(
+            lambda state: self.vehicle.set_relay(Relay.LIGHTS, state))
+
+        if self.controller is not None:
+            self.controller.register_relay_callback(Relay.PVC_FRONT, self.main_tab.widgets.front_deployer_button.toggle)
+            self.controller.register_relay_callback(Relay.CLAW_FRONT, self.main_tab.widgets.front_claw_button.toggle)
+            self.controller.register_relay_callback(Relay.PVC_BACK, self.main_tab.widgets.back_deployer_button.toggle)
+            self.controller.register_relay_callback(Relay.CLAW_BACK, self.main_tab.widgets.back_claw_button.toggle)
+            self.controller.register_relay_callback(Relay.MAGNET, self.main_tab.widgets.magnet_button.toggle)
+            self.controller.register_relay_callback(Relay.LIGHTS, self.main_tab.widgets.lights_button.toggle)
+        
+        self.vehicle.mode_signal.connect(self.main_tab.widgets.manual_button.on_mode)
+        self.main_tab.widgets.manual_button.set_mode_signal(self.vehicle.set_mode_signal)
+        self.vehicle.mode_signal.connect(self.main_tab.widgets.stabilize_button.on_mode)
+        self.main_tab.widgets.stabilize_button.set_mode_signal(self.vehicle.set_mode_signal)
+        self.vehicle.mode_signal.connect(self.main_tab.widgets.depth_hold_button.on_mode)
+        self.main_tab.widgets.depth_hold_button.set_mode_signal(self.vehicle.set_mode_signal)
```

Comment:
I'm not a fan of this, but I haven't thought of a better way to do this yet.

Reply:
I don't disagree